sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/CredentialsRetriever.cs

```diff
@@ -89,13 +76,15 @@ public override void InvokeSync(IExecutionContext executionContext)
         public override async System.Threading.Tasks.Task<T> InvokeAsync<T>(IExecutionContext executionContext)
         {
             ImmutableCredentials ic = null;
-            if (Credentials != null && !(Credentials is AnonymousAWSCredentials))
+            var identity = executionContext.RequestContext.Identity as AWSCredentials;
+
+            if (identity != null && !(identity is AnonymousAWSCredentials))
```

Comment:
It should be even simpler than that, we can just return `null` for `GetCredentials` and `GetCredentialsAsync` has an implementation in the base class that just uses `GetCredentials` and we don't need to change that.

Now this made me think that we shouldn't need the `BearerTokenSigner` special case too that we have above, since the signer cannot be `BearerTokenSigner` when the Identity is `AWSCredentials`.
![image](https://github.com/user-attachments/assets/89720368-99de-469d-9a7a-05972db219ba)